File: VB/SpreadsheetSpellchecking/Form1.vb
Imports System.Collections.Generic
Imports System.Drawing
Imports DevExpress.XtraBars
Imports DevExpress.XtraBars.Ribbon
Imports DevExpress.XtraSpellChecker
Imports DevExpress.XtraSpellChecker.Native
Imports DevExpress.XtraSpreadsheet.Internal

Namespace SpreadsheetSpellchecking

    Public Partial Class Form1
        Inherits RibbonForm

        Private spellChecker As SpellChecker = New SpellChecker()

        Private popupMenu As PopupMenu = New PopupMenu()

        Public Sub New()
            InitializeComponent()
            SpellCheckTextControllersManager.[Default].RegisterClass(GetType(TextBoxWithTransparency), GetType(SimpleTextBoxTextController))
            SpellCheckTextBoxBaseFinderManager.[Default].RegisterClass(GetType(TextBoxWithTransparency), GetType(TextBoxFinder))
            spellChecker.SpellCheckMode = SpellCheckMode.AsYouType
            popupMenu.Manager = BarManager
            AddHandler Me.BarManager.QueryShowPopupMenu, AddressOf OnBarManager_QueryShowPopupMenu
            AddHandler Me.spreadsheetControl.CellEditorOpened, AddressOf OnSpreadsheetControl_CellEditorOpened
        End Sub

        Private ReadOnly Property BarManager As BarManager
            Get
                Return ribbonControl1.Manager
            End Get
        End Property

        Private Sub OnSpreadsheetControl_CellEditorOpened(ByVal sender As Object, ByVal e As DevExpress.XtraSpreadsheet.CellEditorOpenedEventArgs)
            If Not e.IsCustom Then
                BarManager.SetPopupContextMenu(e.Editor, popupMenu)
                spellChecker.Check(e.Editor)
            End If
        End Sub

        Private Sub OnBarManager_QueryShowPopupMenu(ByVal sender As Object, ByVal e As QueryShowPopupMenuEventArgs)
            Dim position As Point = e.Control.PointToClient(e.Position)
            Dim [error] As DevExpress.XtraSpellChecker.Rules.SpellCheckErrorBase = spellChecker.CalcError(position)
            Dim commands As List(Of SpellCheckerCommand) = spellChecker.GetCommandsByError([error])
            If commands Is Nothing Then
                e.Cancel = True
                Return
            End If

            Dim itemLinks As BarItemLinkCollection = popupMenu.ItemLinks
            popupMenu.BeginUpdate()
            Try
                itemLinks.Clear()
                For Each command As SpellCheckerCommand In commands
                    Dim item As BarButtonItem = New BarButtonItem(BarManager, command.Caption)
                    item.Enabled = command.Enabled
                    item.Tag = command
                    AddHandler item.ItemClick, AddressOf OnPopupMenu_ItemClick
                    itemLinks.Add(item)
                Next

                Dim itemShowSpellingForm As BarButtonItem = New BarButtonItem(BarManager, "Show Spelling Form")
                AddHandler itemShowSpellingForm.ItemClick, AddressOf OnPopupMenuShowSpellingForm_ItemClick
                itemLinks.Add(itemShowSpellingForm)
            Finally
                popupMenu.EndUpdate()
            End Try
        End Sub

        Private Sub OnPopupMenu_ItemClick(ByVal sender As Object, ByVal e As ItemClickEventArgs)
            Dim checkerCommand = TryCast(e.Item.Tag, SpellCheckerCommand)
            checkerCommand?.DoCommand()
        End Sub

        Private Sub OnPopupMenuShowSpellingForm_ItemClick(ByVal sender As Object, ByVal e As ItemClickEventArgs)
            spellChecker.Check(Me.ActiveControl)
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles MyBase.Load
            Dim activeSheet = spreadsheetControl.ActiveWorksheet
            activeSheet.Columns(1).WidthInPixels = 150
            spreadsheetControl.SelectedCell = activeSheet("B2")
            spreadsheetControl.SelectedCell.Value = "Missspelled wods"
        End Sub
    End Class
End Namespace

File: CS/SpreadsheetSpellchecking/Form1.cs
using System.Collections.Generic;
using System.Drawing;
using DevExpress.XtraBars;
using DevExpress.XtraBars.Ribbon;
using DevExpress.XtraSpellChecker;
using DevExpress.XtraSpellChecker.Native;
using DevExpress.XtraSpreadsheet.Internal;

namespace SpreadsheetSpellchecking {
    public partial class Form1 : RibbonForm {
        SpellChecker spellChecker = new SpellChecker();
        PopupMenu popupMenu = new PopupMenu();

        public Form1() {
            InitializeComponent();
            SpellCheckTextControllersManager.Default.RegisterClass(typeof(TextBoxWithTransparency), typeof(SimpleTextBoxTextController));
            SpellCheckTextBoxBaseFinderManager.Default.RegisterClass(typeof(TextBoxWithTransparency), typeof(TextBoxFinder));

            spellChecker.SpellCheckMode = SpellCheckMode.AsYouType;

            popupMenu.Manager = BarManager;
            BarManager.QueryShowPopupMenu += OnBarManager_QueryShowPopupMenu;

            spreadsheetControl.CellEditorOpened += OnSpreadsheetControl_CellEditorOpened;
        }

        BarManager BarManager => ribbonControl1.Manager;

        void OnSpreadsheetControl_CellEditorOpened(object sender, DevExpress.XtraSpreadsheet.CellEditorOpenedEventArgs e) {
            if (!e.IsCustom) {
                BarManager.SetPopupContextMenu(e.Editor, popupMenu);
                spellChecker.Check(e.Editor);
            }
        }

        void OnBarManager_QueryShowPopupMenu(object sender, QueryShowPopupMenuEventArgs e) {
            Point position = e.Control.PointToClient(e.Position);
            DevExpress.XtraSpellChecker.Rules.SpellCheckErrorBase error = spellChecker.CalcError(position);
            List<SpellCheckerCommand> commands = spellChecker.GetCommandsByError(error);
            if (commands == null) {
                e.Cancel = true;
                return;
            }

            BarItemLinkCollection itemLinks = popupMenu.ItemLinks;
            popupMenu.BeginUpdate();
            try {
                itemLinks.Clear();
                foreach (SpellCheckerCommand command in commands) {
                    BarButtonItem item = new BarButtonItem(BarManager, command.Caption);
                    item.Enabled = command.Enabled;
                    item.Tag = command;
                    item.ItemClick += OnPopupMenu_ItemClick;
                    itemLinks.Add(item);
                }
                BarButtonItem itemShowSpellingForm = new BarButtonItem(BarManager, "Show Spelling Form");
                itemShowSpellingForm.ItemClick += OnPopupMenuShowSpellingForm_ItemClick;
                itemLinks.Add(itemShowSpellingForm);
            } finally {
                popupMenu.EndUpdate();
            }
        }
        void OnPopupMenu_ItemClick(object sender, ItemClickEventArgs e) {
            var checkerCommand = e.Item.Tag as SpellCheckerCommand;
            checkerCommand?.DoCommand();
        }
        void OnPopupMenuShowSpellingForm_ItemClick(object sender, ItemClickEventArgs e) {
            spellChecker.Check(this.ActiveControl);
        }

        private void Form1_Load(object sender, System.EventArgs e)
        {
            var activeSheet = spreadsheetControl.ActiveWorksheet;
            activeSheet.Columns[1].WidthInPixels = 150;
            spreadsheetControl.SelectedCell = activeSheet["B2"];
            spreadsheetControl.SelectedCell.Value = "Missspelled wods";
        }
    }
}